crates/bevy_color/src/hsla.rs

```diff
@@ -66,6 +66,33 @@ impl Hsla {
     pub const fn with_lightness(self, lightness: f32) -> Self {
         Self { lightness, ..self }
     }
+
+    /// Generate a deterministic but [quasi-randomly distributed](https://en.wikipedia.org/wiki/Low-discrepancy_sequence)
+    /// color from a provided `index`.
+    ///
+    /// These colors are designed to be distributed evenly but randomly through color space.
+    /// This can be helpful for generating debug colors.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bevy_color::Hsla;
+    /// // Palette with 5 distinct hues
+    /// let palette = Hsla::sequence_dispersed().take(5).collect::<Vec<_>>();
+    /// ```
+    pub fn sequence_dispersed() -> impl Iterator<Item = Self> {
+        const FRAC_U32MAX_GOLDEN_RATIO: u32 = 2654435769; // (u32::MAX / Φ) rounded up
+        const RATIO_360: f32 = 360.0 / u32::MAX as f32;
+
+        // from https://extremelearning.com.au/unreasonable-effectiveness-of-quasirandom-sequences/
+        //
+        // Map a sequence of integers (eg: 154, 155, 156, 157, 158) into the [0.0..1.0] range,
+        // so that the closer the numbers are, the larger the difference of their image.
+        (0..=u32::MAX)
```

Comment:
Personally I'm not a fan of the iterator API use here. I think it makes more sense to accept an `u32` and return a `Self`. If someone wants a sequence, they can always do `(0..=u32::MAX).map(Color::from_index)`

Reply:
I agree here that this API can be simplified for better

Reply:
I personally prefer the iterator version, but I also agree it's much cleaner to make the non-iterating version into the iterating one than vice versa, I've just pushed up a change to address this.